url2b9f720f64_jfimbett_QBnB_tgz-pJ8-tests_TestJ8/src/model/QBnBEnv.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * 
+ */
+package model;
+import algorithms.*;
+
+import java.util.ArrayList;
+
+/*************************************************************************
+ *  
+ *  Class: QBnBEnv.java
+ *
+ *  Represents the main optimization environment which will connect directly to the optimization algorithms
+ *
+ *  Author: Juan Imbett
+ *  Email: jfimbett@gmail.com
+ *
+ *************************************************************************/
+public class QBnBEnv {
+
+	/**
+	 * Optimization Model
+	 */
+	private Model m;
+	
+	private double bestSolution;
+	
+	/**
+	 * Parameter that indicates if the solver will print results
+	 */
+	private boolean print;
+	
+	private boolean debugMode;
+	
+	
+	/**
+	 * Creates a new environment
+	 * @param m: Main model
+	 */
+	public QBnBEnv(Model m) {
+		this.m=m;
+	}
+	
+	public void preSolve(){
+		double init=System.currentTimeMillis();
+		m.preSolve();
+		m.update();
+		System.out.println("Presolve time: "+(System.currentTimeMillis()-init)/1000+" seconds.");
+	}
+	/**
+	 * Maximizes the main problem
+	 */
+	public void maximize(){
+		
+		//Identifies the type of the problem
+		if(isMIP()){
+			if(print){
+				printInitial();
+				int columns= m.getConstraints().size();
+				int rows=m.getVariables().size();
+				System.out.println("Solving problem with "+columns+" columns, and "+rows+" rows.");
+				
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 57:63, 85:90.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.